core/logger/audit/audit_logger_test.go

```diff
@@ -0,0 +1,127 @@
+package audit_test
+
+import (
+	"encoding/json"
+	"flag"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/internal/cltest"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/logger/audit"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/urfave/cli"
+)
+
+type MockedHTTPEvent struct {
+	body string
+}
+
+type MockHTTPClient struct {
+	audit.HTTPAuditLoggerInterface
+
+	loggingChannel chan MockedHTTPEvent
+}
+
+type LoginData struct {
+	Email string `json:"email"`
+}
+
+type LoginLogItem struct {
+	EventID string    `json:"eventID"`
+	Env     string    `json:"env"`
+	Data    LoginData `json:"data"`
+}
+
+func (mock *MockHTTPClient) Do(req *http.Request) (*http.Response, error) {
+	b, err := io.ReadAll(req.Body)
+
+	if err != nil {
+		return nil, err
+	}
+
+	message := MockedHTTPEvent{
+		body: string(b),
+	}
+
+	mock.loggingChannel <- message
+
+	return &http.Response{}, nil
+}
+
+func getAuditLoggerConfig() *audit.AuditLoggerConfig {
+	forwardToUrl := "empty"
+	environment := "test"
+	jsonWrapperKey := ""
+
+	return &audit.AuditLoggerConfig{
+		ForwardToUrl:   &forwardToUrl,
+		Environment:    &environment,
+		JsonWrapperKey: &jsonWrapperKey,
+	}
+}
+
+func TestCheckLoginAuditLog(t *testing.T) {
+	t.Parallel()
+
+	// Create a channel that will be used instead of an HTTP client
+	loggingChannel := make(chan MockedHTTPEvent, 2048)
+
+	// Create the mock structure that will be used
+	mockHTTPClient := MockHTTPClient{
+		loggingChannel: loggingChannel,
+	}
+
+	// Create a test logger because the audit logger relies on this logger
+	// as well
+	logger := logger.TestLogger(t)
+
+	// Create new AuditLoggerService
+	auditLogger, err := audit.NewAuditLogger(logger, getAuditLoggerConfig())
+	assert.NoError(t, err)
+
+	// Cast to concrete type so we can swap out the internals
+	auditLoggerService, ok := auditLogger.(*audit.AuditLoggerService)
+	assert.True(t, ok)
+
+	// Swap the internals with a testing handler
+	auditLoggerService.SetLoggingClient(&mockHTTPClient)
+	assert.NoError(t, auditLoggerService.Ready())
+
+	// Create a new chainlink test application passing in our test logger
+	// and audit logger
+	app := cltest.NewApplication(t, logger, auditLogger)
+	require.NoError(t, app.Start(testutils.Context(t)))
```

Comment:
This gives more coverage but is probably overkill, I'd suggest testing the audit logger in isolation, and if you want to test actual audit logging points, do that in a separate integration test using a mock audit logger object.

Reply:
I do like the base of having better coverage. We are planning to expand the Audit implementation in the future so having more coverage will be better.